### PR TITLE
Double tapped event args

### LINKED ISSUE #3240: DoubleTapped event does not give details of where the double tap occurred
It is not obvious  for both DoubleTapped and Tapped events how to get the position of the tap.

### LINKED ISSUE #3756: Porting DoubleTappedRoutedEventArgs
The WPF `DoubleTapped` event hands out `DoubleTappedRoutedEventArgs` that allows you to get the current position. Our `DoubleTapped` merely gives a `RoutedEventArgs`. It would be great to port the behavior from WPF.

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?
DoubleTapped, Tapped, RightTapped events now have more specific arguments with GetPostiion method.
See:
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.doubletappedroutedeventargs?view=winrt-19041
https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.tappedroutedeventargs?view=winrt-19041
https://docs.microsoft.com/en-us/uwp/api/windows.ui.input.righttappedeventargs?view=winrt-19041

Note that RightTapped event args in UWP has Position property instead of GetPosition(relativeTo) method, so it's not 1to1 port.

## Breaking changes
Event args types were changed, so it can be considered as a binary breaking change.

## Fixed issues
Fixes https://github.com/AvaloniaUI/Avalonia/issues/3756
Fixes https://github.com/AvaloniaUI/Avalonia/issues/3240

### CODE REVIEW

File: src/Avalonia.Input/TappedGestureEventArgs.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Input
+{
+    public class DoubleTappedEventArgs : RoutedEventArgs
```
Comment (kekekeks) [inline src/Avalonia.Input/TappedGestureEventArgs.cs:6]: Why do we have 3 separate classes instead of one?

Comment (maxkatz6) [reply src/Avalonia.Input/TappedGestureEventArgs.cs:6]: No particular reason, just using nearly same API with different classes as in UWP.
As for me it would be also fine to have single TappedEventArgs.

Comment (robloo) [reply src/Avalonia.Input/TappedGestureEventArgs.cs:6]: It seems Microsoft has the mentality of code duplication in a lot of places in UWP. I've had similar discussions in the WinUI repo for new controls where they were making brand-new enums and similar properties instead of re-using or tweaking existing concepts.

Aside from not thinking things through enough to reach a good level of code-reuse, I think they do this to ensure they aren't locked in if they need to change something for one control but not another. It's a direct result of Microsoft's policy of no breaking changes. Personally, I miss the WPF mindset where they unified everything possible and you could see evidence of that throughout the entire framework. Everything was more logical and easier to remember.

Long story short, I am a fan of not repeating Microsoft's mistakes and searching for the best possible API design is quite important. Of course it must be balanced with WinUI code compatiblity. In this case we might be able to have both, at least from a framework standpoint.

What about making the common `TappedEventArgs` the base class for the other two?

```csharp
public class TappedEventArgs : RoutedEventArgs
public class DoubleTappedEventArgs : TappedEventArgs 
public class RightTappedEventArgs  : TappedEventArgs
```

Comment (maxkatz6) [reply src/Avalonia.Input/TappedGestureEventArgs.cs:6]: > Of course it must be balanced with WinUI code compatiblity

From that side we already is off from that, because in WinUI this events called DoubleTappedRoutedEventArgs / TappedRoutedEventArgs. See @grokys comment on previous PR for this API - https://github.com/AvaloniaUI/Avalonia/pull/3763#issuecomment-611441473

I will update this PR to use single TappedEventArgs for now. And I have doubts that we will need something specific for Double/Right tapped event args in the future.

**Comment by kekekeks** [comment]
@maxkatz6 can we keep RoutedEvent<RoutedEventArgs> signature as is but actually fire the event with new event args? That would allow us to keep binary compatibility with 0.10.0 and still get the feature in 0.10.1. For 0.11.0 the signature should be updated.

**Comment by grokys** [comment]
>can we keep RoutedEvent<RoutedEventArgs> signature as is but actually fire the event with the new event args? That would allow us to keep binary compatibility with 0.10.0 and still get the feature in 0.10.1. For 0.11.0 the signature should be updated.

Yeah I meant to suggest this actually. Would be a good solution for now.

**Comment by maxkatz6** [comment]
I will update PR later today, good idea

**Review by grokys — APPROVED** [review]
